### ISSUE TITLE
Federation initially working with NC13, then became buggy after upgrade to NC15, then stop working with NC16

### ISSUE THREAD

**Issue by RussianNeuroMancer** [nextcloud/server#15082]
### Steps to reproduce
1. Configure Apache 2 and php-fpm on Ubuntu 18.04.
2. Install Nextcloud 16.0.0RC1.
3. Try to share folder to other Nextcloud server.
4. Try to access shared folder from other Nextcloud server.

### Expected behaviour
Sharing folder works.

### Actual behaviour
Sharing folder works with Nextcloud 13 without issues.

After upgrade NC13 instances to NC14 and then to NC15, sharing folders became buggy with (with exactly same Apache 2 and php-fpm configuration, no changes here at all). For example editing text files inside shared folder works, changes get saved to source server where file is located, but server write error "No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured" to log, and report http code 500 to client. Same problem with transferring files to NC15 server from NC13 or NC15 - file is saved, but client receive error code from server.

After upgrade one NC15 instance to NC16, sharing folders stop working (still with exactly the same Apache 2 and php-fpm configuration). When "client" NC15 or NC13 instance trying to access folder shared from "server" NC16 instance - NC16 write same error to log, but now "client" instances can't even display folder content. **At the same time, access to NC16 via WebDAV,  moving and editing files (for example from Nautilus file manager or GEdit text editor) works fine.** 

Obviously, all instances configured to trust each other, and of course autocomplete usernames and server addresses. Also please check PHP-FPM configuration at the end of this message.

### Server configuration

**Operating system**: Ubuntu 18.04 LTS and Ubuntu 19.04, issue is reproducible in both cases.

**Web server:** 2.4.29 and 2.4.38 was tested.

**Database:** PostgreSQL 10.7

**PHP version:** 7.2.15

**Nextcloud version:** 15.0.5, 16.0.0 RC1.

**Updated from an older Nextcloud/ownCloud or fresh install:** some NC15 instances was upgraded from NC13 fresh installation, some NC15/NC16 instances was upgraded from NC13 that upgraded since project beginning days, but no difference in behaviour was found so far. 

**Where did you install Nextcloud from:** from Nextcloud download page.

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.2.0
  - activity: 2.9.0
  - admin_audit: 1.6.0
  - bruteforcesettings: 1.3.0
  - calendar: 1.7.0
  - cloud_federation_api: 0.2.0
  - comments: 1.6.0
  - contacts: 3.1.0
  - dav: 1.9.2
  - federatedfilesharing: 1.6.0
  - federation: 1.6.0
  - files: 1.11.0
  - files_fulltextsearch: 1.3.0
  - files_pdfviewer: 1.5.0
  - files_rightclick: 0.13.0
  - files_sharing: 1.8.0
  - files_texteditor: 2.8.0
  - files_trashbin: 1.6.0
  - files_versions: 1.9.0
  - firstrunwizard: 2.5.0
  - fulltextsearch: 1.3.0
  - fulltextsearch_elasticsearch: 1.3.0
  - gallery: 18.3.0
  - logreader: 2.1.0
  - lookup_server_connector: 1.4.0
  - nextcloud_announcements: 1.5.0
  - notes: 2.5.1
  - notifications: 2.4.1
  - oauth2: 1.4.2
  - password_policy: 1.6.0
  - privacy: 1.0.0
  - provisioning_api: 1.6.0
  - recommendations: 0.4.0
  - serverinfo: 1.6.0
  - sharebymail: 1.6.0
  - support: 1.0.0
  - survey_client: 1.4.0
  - systemtags: 1.6.0
  - theming: 1.7.0
  - twofactor_backupcodes: 1.5.0
  - updatenotification: 1.6.0
  - viewer: 1.0.0
  - workflowengine: 1.6.0
Disabled:
  - encryption
  - files_external
  - files_videoplayer
  - mail
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "16.0.0.6",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "asset-pipeline.enabled": false,
        "enabledPreviewProviders": {
            "0": "OC\\Preview\\PNG",
            "1": "OC\\Preview\\JPEG",
            "2": "OC\\Preview\\GIF",
            "3": "OC\\Preview\\BMP",
            "4": "OC\\Preview\\XBitmap",
            "5": "OC\\Preview\\MP3",
            "7": "OC\\Preview\\MarkDown",
            "8": "OC\\Preview\\MSOffice2003",
            "9": "OC\\Preview\\MSOffice2007",
            "10": "OC\\Preview\\MSOfficeDoc",
            "11": "OC\\Preview\\OpenDocumen",
            "12": "OC\\Preview\\PDF"
        },
        "maintenance": false,
        "loglevel": 0,
        "updatechecker": false,
        "theme": "",
        "mail_smtpmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "PLAIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/nextcloud",
        "updater.release.channel": "beta",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
Other Nextcloud instances, such as Nextcloud 13.0.4, 15.0.5, 16.0.0RC1.

#### Web server error log
<details>
<summary>Web server error log</summary>

```
[Sat Apr 13 00:00:34.238948 2019] [mpm_worker:notice] [pid 279:tid 140368501267392] AH00292: Apache/2.4.38 (Ubuntu) mod_fcgid/2.3.9 OpenSSL/1.1.1b configured -- resuming normal operations
[Sat Apr 13 00:00:34.238997 2019] [core:notice] [pid 279:tid 140368501267392] AH00094: Command line: '/usr/sbin/apache2'
[Sat Apr 13 00:14:26.122633 2019] [proxy_fcgi:error] [pid 6269:tid 140368189753088] [client 176.215.255.186:39298] AH01071: Got error 'Primary script unknown\n'
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"k57oGFEmAvGKUNRPZ5no","level":0,"time":"2019-04-12T23:49:15+08:00","remoteAddr":"176.215.255.186","user":"--","app":"webdav","method":"PROPFIND","url":"\/nextcloud\/public.php\/webdav\/","message":{"Exception":"Sabre\\DAV\\Exception\\NotAuthenticated","Message":"No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured","Code":0,"Trace":[{"function":"beforeMethod","class":"Sabre\\DAV\\Auth\\Plugin","type":"->","args":[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php","line":105,"function":"call_user_func_array","args":[[{"autoRequireLogin":true,"__class__":"Sabre\\DAV\\Auth\\Plugin"},"beforeMethod"],[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","line":466,"function":"emit","class":"Sabre\\Event\\EventEmitter","type":"->","args":["beforeMethod",[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","line":254,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/nextcloud\/apps\/dav\/appinfo\/v1\/publicwebdav.php","line":107,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"\/var\/www\/nextcloud\/public.php","line":79,"args":["\/var\/www\/nextcloud\/apps\/dav\/appinfo\/v1\/publicwebdav.php"],"function":"require_once"}],"File":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Auth\/Plugin.php","Line":168,"CustomMessage":"--"},"userAgent":"sabre-dav\/3.2.2 (http:\/\/sabre.io\/)","version":"16.0.0.6"}
{"reqId":"pwzizdJtGrGvbrKlSWiZ","level":0,"time":"2019-04-12T23:49:15+08:00","remoteAddr":"176.215.255.186","user":"--","app":"webdav","method":"PROPFIND","url":"\/nextcloud\/public.php\/webdav\/","message":{"Exception":"Sabre\\DAV\\Exception\\NotAuthenticated","Message":"No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured","Code":0,"Trace":[{"function":"beforeMethod","class":"Sabre\\DAV\\Auth\\Plugin","type":"->","args":[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php","line":105,"function":"call_user_func_array","args":[[{"autoRequireLogin":true,"__class__":"Sabre\\DAV\\Auth\\Plugin"},"beforeMethod"],[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","line":466,"function":"emit","class":"Sabre\\Event\\EventEmitter","type":"->","args":["beforeMethod",[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","line":254,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/nextcloud\/apps\/dav\/appinfo\/v1\/publicwebdav.php","line":107,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"\/var\/www\/nextcloud\/public.php","line":79,"args":["\/var\/www\/nextcloud\/apps\/dav\/appinfo\/v1\/publicwebdav.php"],"function":"require_once"}],"File":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Auth\/Plugin.php","Line":168,"CustomMessage":"--"},"userAgent":"sabre-dav\/3.2.2 (http:\/\/sabre.io\/)","version":"16.0.0.6"}
{"reqId":"gA0yxAwCuvdce5v4ujav","level":0,"time":"2019-04-12T23:49:30+08:00","remoteAddr":"176.215.255.186","user":"--","app":"webdav","method":"PROPFIND","url":"\/nextcloud\/public.php\/webdav\/","message":{"Exception":"Sabre\\DAV\\Exception\\NotAuthenticated","Message":"No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured","Code":0,"Trace":[{"function":"beforeMethod","class":"Sabre\\DAV\\Auth\\Plugin","type":"->","args":[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php","line":105,"function":"call_user_func_array","args":[[{"autoRequireLogin":true,"__class__":"Sabre\\DAV\\Auth\\Plugin"},"beforeMethod"],[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","line":466,"function":"emit","class":"Sabre\\Event\\EventEmitter","type":"->","args":["beforeMethod",[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]]},{"file":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php","line":254,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":[{"absoluteUrl":"https:\/\/node1.servername.com\/nextcloud\/public.php\/webdav\/","__class__":"Sabre\\HTTP\\Request"},{"__class__":"Sabre\\HTTP\\Response"}]},{"file":"\/var\/www\/nextcloud\/apps\/dav\/appinfo\/v1\/publicwebdav.php","line":107,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"\/var\/www\/nextcloud\/public.php","line":79,"args":["\/var\/www\/nextcloud\/apps\/dav\/appinfo\/v1\/publicwebdav.php"],"function":"require_once"}],"File":"\/var\/www\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Auth\/Plugin.php","Line":168,"CustomMessage":"--"},"userAgent":"sabre-dav\/3.2.2 (http:\/\/sabre.io\/)","version":"16.0.0.6"}
```
</details>

#### PHP-FPM 7.2.15
<details>
<summary>PHP-FPM 7.2.15 Apache module config</summary>

```
# Redirect to local php-fpm if mod_php is not available
<IfModule !mod_php7.c>
<IfModule proxy_fcgi_module>
    # Enable http authorization headers
    <IfModule setenvif_module>
    SetEnvIfNoCase ^Authorization$ "(.+)" HTTP_AUTHORIZATION=$1
    </IfModule>

    <FilesMatch ".+\.ph(ar|p|tml)$">
        SetHandler "proxy:unix:/run/php/php7.2-fpm.sock|fcgi://localhost"
    </FilesMatch>
    <FilesMatch ".+\.phps$">
        # Deny access to raw php sources by default
        # To re-enable it's recommended to enable access to the files
        # only in specific virtual host or directory
        Require all denied
    </FilesMatch>
    # Deny access to files without filename (e.g. '.php')
    <FilesMatch "^\.ph(ar|p|ps|tml)$">
        Require all denied
    </FilesMatch>
</IfModule>
</IfModule>

```
As you can see this is default PHP-FPM 7.2.15 configration and HTTP_AUTHORIZATION is passed here.

</details>

**Comment by violoncelloCH** [comment]
cc @nextcloud/sharing

**Comment by RussianNeuroMancer** [comment]
> At the same time, access to NC16 via WebDAV, moving and editing files (for example from Nautilus file manager or GEdit text editor) works fine.

After upgrade from 16.0.0.6 to 16.0.0.9 this statement is no longer true. I getting same error on server side, and "500 Internal Server Error" on client side (in gedit). Attempt to replace existing file on Nextcloud via Nautilus also cause similar issue.

**Comment by RussianNeuroMancer** [comment]
It's seems like issue was resolved by either 16.0.1 or 16.0.2.
